### PR TITLE
Sending lang_available in courses response

### PR DESCRIPTION
**Pull request by pysaquib**
1. Reads all properties json file course wise.
2. Adds a new property in course objects called `lang_available` setting its value to an array of all available languages files.

### CODE REVIEW

File: lib/routes/courses.js [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
+const _ = require('lodash');
+const glob = require('glob');
+
 const Courses = require('../models/courses');
 
+const propertiesFileObj = {};
+const propFiles = glob.sync('**/PROPERTIES_FILES/**/*.json');
+if (propFiles.length > 0) {
+  _.map(propFiles, (fileName) => {
+    const name = fileName.split('/').pop().split('_')[0];
```
Comment (saiky76) [inline lib/routes/courses.js:11]: My understanding is name here will give us _hi, _en. Correct?

File: lib/routes/courses.js [hunk 2]
```diff
@@ -1,6 +1,26 @@
 const Joi = require('@hapi/joi');
+const _ = require('lodash');
+const glob = require('glob');
+
 const Courses = require('../models/courses');
 
+const propertiesFileObj = {};
+const propFiles = glob.sync('**/PROPERTIES_FILES/**/*.json');
```
Comment (saiky76) [inline lib/routes/courses.js:8]: Is this searching for property files in the course folders or through out the application?

Comment (pysaquib) [reply lib/routes/courses.js:8]: In the curriculum folder.
- curriculum
  - course_1
     - properties_files
  - course_2
     - properties_files

File: lib/routes/courses.js [hunk 3]
```diff
@@ -1,6 +1,26 @@
...
+if (propFiles.length > 0) {
+  _.map(propFiles, (fileName) => {
+    const name = fileName.split('/').pop().split('_')[0];
+    if (!propertiesFileObj[name]) {
+      propertiesFileObj[name] = [];
+      propertiesFileObj[name].push(fileName.split('/').pop().split('_').pop().split('.')[0]);
```
Comment (saiky76) [inline lib/routes/courses.js:14]: Also can you not recompute this again
 fileName.split('/').pop().split('_').pop().split('.')[0]) 

you can save some section of this already derived above and reuse it here

File: lib/routes/courses.js [hunk 4]
```diff
@@ -1,6 +1,26 @@
 const Joi = require('@hapi/joi');
+const _ = require('lodash');
+const glob = require('glob');
+
 const Courses = require('../models/courses');
 
+const propertiesFileObj = {};
```
Comment (saiky76) [inline lib/routes/courses.js:7]: please rename this as courseLangMap as that is what the purpose of this global map is

**Review by saiky76 — APPROVED** [review]

**Pull request merged**